src/index.ts

```diff
@@ -180,212 +155,208 @@ const getProfileFixture = (p?: Partial<Profile>): Profile =>
-    generateMessageContentFixture()
-  );
-
-  const aMessageStatus = getMessageStatusFixture({
-    messageId: aMessage.id,
-  });
-  const errorOrMessageStatus = await messageStatusModel.create(
-    aMessageStatus,
-    aMessageStatus.messageId
-  );
 
-  const createdMessageStatus = errorOrMessageStatus.getOrElseL(() => {
-    throw new Error("cannot create new message status");
-  });
-  // create a new version
-  await messageStatusModel.update(
-    createdMessageStatus.id,
-    createdMessageStatus.messageId,
-    () => createdMessageStatus
-  );
-
-  const aNotification = getNotificationFixture({
-    messageId: aMessage.id,
-    fiscalCode: aMessage.fiscalCode,
-  });
-  await notificationModel.create(aNotification, aNotification.messageId);
+  await pipe(
+    messageModel.create(aMessage),
+    TE.chainW(() => {
+      /*
+        NOTICE: io-functions-commons is using a deprecated version for the blob storage.
+                here i will use the deprecated library client just to let it works
+      */
+      return messageModel.storeContentAsBlob(
+        createOldBlobService(storageConnectionString),
+        aMessage.id,
+        generateMessageContentFixture()
+      )
+    }),
+    TE.chainW(() => {
+      const aMessageStatus = getMessageStatusFixture({ messageId: aMessage.id });
 
-  const aNotificationStatus = getNotificationStatusFixture({
-    messageId: aMessage.id,
-    notificationId: aNotification.id,
-  });
-  const errorOrNotifcationStatus = await notificationStatusModel.create(
-    aNotificationStatus,
-    aNotificationStatus.notificationId
-  );
-
-  const createdNotificationStatus = errorOrNotifcationStatus.getOrElseL(
-    (err) => {
-      throw new Error("cannot create new notification status:" + toString(err));
-    }
-  );
-  // create a new version
-  await notificationStatusModel.update(
-    createdNotificationStatus.id,
-    createdNotificationStatus.notificationId,
-    () => createdNotificationStatus
-  );
+      return messageStatusModel.create(
+        aMessageStatus
+      )
+    }),
+    //create a new version
+    TE.chainW(createdMessageStatus =>
+      messageStatusModel.update(
+        createdMessageStatus
+      )
+    ),
+    TE.bindW("aNotification", () => TE.of(getNotificationFixture({ messageId: aMessage.id, fiscalCode: aMessage.fiscalCode }))),
+    TE.chainFirstW(({ aNotification }) => notificationModel.create(aNotification)),
+    TE.bindW("aNotificationStatus", ({ aNotification }) => TE.of(getNotificationStatusFixture({ messageId: aMessage.id, notificationId: aNotification.id }))),
+    TE.bindW("createdNotificationStatus", ({ aNotificationStatus }) => notificationStatusModel.create(aNotificationStatus)),
+    TE.mapLeft((err) => {
+      throw new Error("Cannot create new notification status" + err);
+    }),
+    TE.chainFirstW(({ createdNotificationStatus }) => notificationStatusModel.update(createdNotificationStatus))
+  )()
 
-  return aMessage.id;
+  return aMessage.id
```

Comment:
I'd prefer the version with the trailing ";". I may remember that our linting rules add all missing ";", I don't know why it's not an issue here. Maybe we are not linting this code.

Reply:
yes, the code here is not linted. However it will be fixed after merging this and #13